### PR TITLE
fix(component_related_blocks): move shared logic to base cell

### LINKED ISSUE #20: Add admin specs

### PR DESCRIPTION
**Pull request by verarojman**
- Improve consistency for blocks related to components
- Add spec and lint base cell
- Add component scope to upcoming meetings cell
- Use set manifest_name in component-related cells

Closes #20 

Closes #30

### CODE REVIEW

File: app/cells/decidim/alternative_landing/content_blocks/base_cell.rb [hunk 1]
```diff
@@ -18,14 +18,22 @@ def participatory_spaces
...
           end.unshift [t(".all"), nil]
         end
 
         def component
-          @component ||= Decidim::Component.find_by(id: form.object.settings.try(:component_id))
+          components.find_by(id: (defined?(form) ? form.object : model).settings.try(:component_id))
+        end
+
+        def components
+          @components ||= Decidim::Component.where(participatory_space: participatory_spaces)
```
Comment (microstudi) [inline app/cells/decidim/alternative_landing/content_blocks/base_cell.rb:32]:
```suggestion
          Decidim::Component.where(participatory_space: participatory_spaces)
```
Not sure memoize does anything here

File: app/cells/decidim/alternative_landing/content_blocks/base_cell.rb [hunk 2]
```diff
@@ -18,14 +18,22 @@ def participatory_spaces
...
-          Decidim::Component.published.where(participatory_space: participatory_spaces, manifest_name: manifest_name).map do |component|
+        def available_components
+          components.where(manifest_name: manifest_name).map do |component|
             ["#{translated_attribute(component.name)} (#{translated_attribute(component.participatory_space.title)})", component.id]
           end.unshift [t(".all"), nil]
         end
 
         def component
-          @component ||= Decidim::Component.find_by(id: form.object.settings.try(:component_id))
+          components.find_by(id: (defined?(form) ? form.object : model).settings.try(:component_id))
```
Comment (microstudi) [inline app/cells/decidim/alternative_landing/content_blocks/base_cell.rb:28]:
```suggestion
          @component ||= components.find_by(id: (defined?(form) ? form.object : model).settings.try(:component_id))
```
Maybe?

File: app/cells/decidim/alternative_landing/content_blocks/base_cell.rb [hunk 3]
```diff
@@ -18,14 +18,22 @@ def participatory_spaces
           ].flatten.compact
         end
 
-        def available_components(manifest_name)
-          Decidim::Component.published.where(participatory_space: participatory_spaces, manifest_name: manifest_name).map do |component|
+        def available_components
+          components.where(manifest_name: manifest_name).map do |component|
```
Comment (microstudi) [inline app/cells/decidim/alternative_landing/content_blocks/base_cell.rb:22]:
```suggestion
          @available_components ||= components.where(manifest_name: manifest_name).map do |component|
```
maybe memoize here this mapping

**Review by microstudi — COMMENTED** [review]
I'd only change some memoization patterns

File: app/cells/decidim/alternative_landing/content_blocks/base_cell.rb (hunk 1 above)
Comment (verarojman) [reply app/cells/decidim/alternative_landing/content_blocks/base_cell.rb:32]: Does it hurt, though? 🤔

File: app/cells/decidim/alternative_landing/content_blocks/base_cell.rb (hunk 2 above)
Comment (verarojman) [reply app/cells/decidim/alternative_landing/content_blocks/base_cell.rb]: Sure, thanks!

File: app/cells/decidim/alternative_landing/content_blocks/base_cell.rb (hunk 3 above)
Comment (verarojman) [reply app/cells/decidim/alternative_landing/content_blocks/base_cell.rb]: Great :D

File: app/cells/decidim/alternative_landing/content_blocks/base_cell.rb (hunk 1 above)
Comment (microstudi) [reply app/cells/decidim/alternative_landing/content_blocks/base_cell.rb:32]: sure not

**Review by fblupi — APPROVED** [review]

**Pull request merged**